chainerx_cc/chainerx/CMakeLists.txt

```diff
@@ -84,20 +84,34 @@ add_library(chainerx SHARED
     thread_local_state.cc
     )
 
-target_link_libraries(chainerx
-    PUBLIC
-    ${CMAKE_DL_LIBS}
-    PRIVATE
-    -Wl,--whole-archive)
+if(${APPLE})
+  target_link_libraries(chainerx
+      PUBLIC
+      ${CMAKE_DL_LIBS}
+      PRIVATE
+      -Wl)
```

Comment:
Is it okay to skip `--whole-archive` instead of using e.g. `-force_load` or `-all_load`?

Reply:
I'm not too bright on the subject so I will have to do some research. It would be great if there is somebody who can point out a workaround for this.